src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -519,6 +520,10 @@ protected GraphQLFieldDefinition getField(Method method) throws GraphQLAnnotatio
                     if (graphQLType instanceof GraphQLObjectType) {
                         GraphQLInputObjectType inputObject = getInputObject((GraphQLObjectType) graphQLType);
                         graphQLType = inputObject;
+                    } else if (graphQLType instanceof GraphQLModifiedType &&
```

Comment:
This needs a unwrap stack.  A type can be a non null list of list of non null types or even deeper
eg:

[ [ type ! ] ! ] !

The way to do this is via a stack.  unwrap the type all the way to the bottom